trussworks/react-uswds
# ci: build artifact using CI

# Summary

Adds CI to build the tgz artifact so that local developer machines are not used, which can cause sporadic problems if things are not _pristine_ (see #1725)

## Related Issues or PRs

Closes #1725 

## How To Test

Since this action runs when a PR titled `release-*` is merged, it's certainly hard to test at this stage. I allowed the workflow to run on this PR by commenting that logic, to verify the meat of the functionality actually works. Workflow link here: https://github.com/trussworks/react-uswds/actions/runs/1412820829

Sadly, [there is no way to get upload-artifact to not zip the already zipped `.tgz`](https://github.com/actions/upload-artifact#zipped-artifact-downloads), so we have this annoying manual step to unzip it 🤷 

### Screenshots

Workflow page:
![image](https://user-images.githubusercontent.com/15805554/139908010-ca7da7f2-15d5-49a1-b084-a329a848d3aa.png)

Download `Artifact.zip` and unzip to get the `.tgz`:
![image](https://user-images.githubusercontent.com/15805554/139908286-b6791cfb-9452-471a-b623-f81cff9d1ef5.png)
